content/docs/command-reference/add.md

```diff
@@ -11,57 +10,46 @@ usage: dvc add [-h] [-q | -v] [-f] [--no-commit]
+already tracked dataset. If the target is a part of an existing dataset, only
+that portion of the dataset is updated, and associated `.dvc` file is updated to
+reflect the changes. If the target path does not exist in workspace but was
+previously tracked, it will be removed from the dataset, and the `.dvc` file
+will be updated accordingly. For new files and
+[directories](#adding-entire-directories) that are not currently tracked, DVC
+creates new `.dvc` files to track the added data, and stores them in the
+<abbr>cache</abbr>.
+
+Leveraging the metadata in `.dvc` files and the <abbr>cache</abbr> structure,
+datasets don't need to exist completely in your workspace to update. You can
+pull a dataset partially and operate on it. DVC will automatically update the
+relevant `.dvc` file to reflect the changes. This capability allows you to
+[work with large datasets without having to download the entire dataset](/docs/user-guide/data-management/modifying-large-datasets#modifying-remote-datasets)
+to your local machine.
 
 This command can be used to track large files, models, dataset directories, etc.
 that are too big for Git to handle directly. This enables
 [versioning](/doc/use-cases/versioning-data-and-models) them indirectly with
 Git.
 
-The `targets` are the files or [directories](#adding-entire-directories) to add.
-They get stored in the <abbr>cache</abbr> by default (use the `--no-commit`
-option to avoid this, and `dvc commit` to finish the process when needed).
-
 > See also `dvc.yaml` and `dvc stage add` for more advanced ways to track and
 > version intermediate and final results (like ML models).
 
-After checking that each `target` hasn't been added before (or tracked with
-other DVC commands), a few actions are taken under the hood:
-
-1. Calculate the file hash.
-2. Move the file contents to the cache (by default in `.dvc/cache`) (or to
-   [remote storage] if `--to-remote` is given), using the file hash to form the
-   cached file path. (See [Structure of cache directory] for more details.)
-3. Attempt to replace the file with a link to the cached data (more details on
-   file linking further down). Skipped if `--to-remote` is used.
-4. Create a corresponding `.dvc` file to track the file, using its path and hash
-   to identify the cached data (with `--to-remote`/`-o`, an external path is
-   moved to the <abbr>workspace</abbr>). The `.dvc` file lists the DVC-tracked
-   file as an <abbr>output</abbr> (`outs` field). The `.dvc` file name generated
-   is `<file>.dvc`, where `<file>` is the file name of the first target.
-5. Add the `targets` to `.gitignore` in order to prevent them from being
-   committed to the Git repository (unless `dvc init --no-scm` was used when
-   initializing the <abbr>DVC project</abbr>).
-6. Instructions are printed showing `git` commands for staging `.dvc` files (or
-   they are staged automatically if [`core.autostage`] is set).
```

Comment:
Do you think we need to keep any of these mechanics? I wonder if we should at least mention that for new files, the actual paths are added to `.gitignore`. What do you think?

Reply:
It felt like too much to me. I'll reintroduce about `.gitignore` as it feels important. 
Happy to revert if you disagree though.

Reply:
No, I agree with the overall direction. `.gitignore` was the only glaring omission.